File: persistence/src/main/java/io/sparkled/persistence/playlist/PlaylistPersistenceService.java
package io.sparkled.persistence.playlist;

import io.sparkled.model.entity.Playlist;
import io.sparkled.model.entity.PlaylistSequence;
import io.sparkled.model.entity.PlaylistSummary;
import io.sparkled.model.entity.Sequence;

import java.util.List;
import java.util.Map;
import java.util.Optional;
import java.util.UUID;

public interface PlaylistPersistenceService {

    Playlist createPlaylist(Playlist playlist);

    List<Playlist> getAllPlaylists();

    Map<Integer, PlaylistSummary> getPlaylistSummaries();

    Optional<Playlist> getPlaylistById(int playlistId);

    Optional<Sequence> getSequenceAtPlaylistIndex(int playlistId, int index);

    List<PlaylistSequence> getPlaylistSequencesByPlaylistId(int playlistId);

    Optional<PlaylistSequence> getPlaylistSequenceByUuid(int sequenceId, UUID uuid);

    void savePlaylist(Playlist playlist, List<PlaylistSequence> playlistSequences);

    void deletePlaylist(int playlistId);
}

File: persistence/src/main/kotlin/io/sparkled/persistence/playlist/PlaylistPersistenceService.kt
package io.sparkled.persistence.playlist

import io.sparkled.model.entity.Playlist
import io.sparkled.model.entity.PlaylistSequence
import io.sparkled.model.entity.Sequence
import io.sparkled.model.playlist.PlaylistSummary
import java.util.UUID

interface PlaylistPersistenceService {

    fun createPlaylist(playlist: Playlist): Playlist

    fun getAllPlaylists(): List<Playlist>

    fun getPlaylistSummaries(): Map<Int, PlaylistSummary>

    fun getPlaylistById(playlistId: Int): Playlist?

    fun getSequenceAtPlaylistIndex(playlistId: Int, index: Int): Sequence?

    fun getSequencesByPlaylistId(playlistId: Int): List<Sequence>

    fun getPlaylistSequencesByPlaylistId(playlistId: Int): List<PlaylistSequence>

    fun getPlaylistSequenceByUuid(sequenceId: Int, uuid: UUID): PlaylistSequence?

    fun savePlaylist(playlist: Playlist, playlistSequences: List<PlaylistSequence>)

    fun deletePlaylist(playlistId: Int)

    fun getPlaylistNames(): Map<Int, String>
}
